org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CodeLensHandler.java

```diff
@@ -113,7 +122,7 @@ public void acceptSearchMatch(SearchMatch match) throws CoreException {
 	}
 
 	public List<CodeLens> getCodeLensSymbols(String uri, IProgressMonitor monitor) {
-		if (!preferenceManager.getPreferences().isReferencesCodeLensEnabled()) {
+		if (!preferenceManager.getPreferences().isReferencesCodeLensEnabled() && !preferenceManager.getPreferences().isImplementationsCodeLensEnabled()) {
```

Comment:
we should probably have preferences.isCodeLensEnabled {return isReferencesCodeLensEnabled || isImplementationsCodeLensEnabled}